### PR TITLE
More restyling done on my own scripts.

### PR DESCRIPTION
**Pull request by Gyrth**
Related to #50

### CODE REVIEW

File: assets/RocketDisplay/CameraControl.gd [hunk 1]
```diff
@@ -1,28 +1,31 @@
...
-func _physics_process(delta):
+export (float) var max_zoom_distance : float = 50.0
+export (float) var min_zoom_distance : float = 2.5
+export (float) var zoom_step_size : float = 0.5
+export (float, 0, 500) var speed : float = 5
+export (NodePath) var camera_path : NodePath
+
+onready var camera : Node = get_node(camera_path)
+
+const MOUSE_SENSITIVITY : float = 0.4
```
Comment (TheDuriel) [inline assets/RocketDisplay/CameraControl.gd]: can we move constants above onready vars?

File: assets/RocketDisplay/CameraControl.gd [hunk 2]
```diff
@@ -1,28 +1,31 @@
...
+var target_rotation_degrees : Vector3 = Vector3()
+var look_direction : Vector2 = Vector2()
+var mouse_down : bool = false
+
+func _physics_process(delta) -> void:
 	if not enabled:
 		return
-	
+
 	$CameraPosition.translation = $CameraPosition.translation.linear_interpolate(Vector3(0, 0, current_zoom_distance), delta * speed)
```
Comment (TheDuriel) [inline assets/RocketDisplay/CameraControl.gd]: these paths should be chached in an onready var

File: assets/RocketDisplay/CameraControl.gd [hunk 3]
```diff
@@ -32,47 +35,47 @@ func _physics_process(delta):
...
 		current_zoom_distance = $CameraPosition.translation.z
 		look_direction.y = rotation_degrees.x
 		look_direction.x = rotation_degrees.y
 		target_rotation_degrees = Vector3(look_direction.y, look_direction.x, 0)
-	
+
 	enabled = _enabled
 
-func _input(event):
+func _input(event) -> void:
```
Comment (TheDuriel) [inline assets/RocketDisplay/CameraControl.gd]: event: InputEvent

will improve auto complete ;P

File: assets/RocketDisplay/RocketDisplay.gd [hunk 4]
```diff
@@ -1,112 +1,103 @@
...
-
-export(NodePath) var camera_path
-onready var camera = get_node(camera_path)
-
-func _ready():
+export(NodePath) var camera_path : NodePath
+
+onready var camera : Node = get_node(camera_path)
+
+const MAX_STAGES : int = 5
```
Comment (TheDuriel) [inline assets/RocketDisplay/RocketDisplay.gd]: same thing with the constants

File: assets/RocketDisplay/RocketDisplay.gd [hunk 5]
```diff
@@ -1,112 +1,103 @@
...
+const TRANSITION_DURATION : float = 1.0
+
+var current_stage : int = 1
+var transition_timer : float = TRANSITION_DURATION
+var transition_transform_from : Transform
+var transition_transform_to_object : Node
+var return_camera : Node
+
+func _ready() -> void:
 	$Control.visible = false
```
Comment (TheDuriel) [inline assets/RocketDisplay/RocketDisplay.gd]: Paths need caching in onready vars

File: assets/RocketDisplay/RocketDisplay.gd [hunk 6]
```diff
@@ -1,112 +1,103 @@
...
 #	yield(get_tree(), "idle_frame")
-#	Activate()
+#	activate()
 
-func RegisterInfoBox(var node):
-	info_boxes.append(node)
-
-func _process(delta):
-	if transition_timer < transition_duration:
+func _process(delta) -> void:
```
Comment (TheDuriel) [inline assets/RocketDisplay/RocketDisplay.gd]: delta: float

File: assets/RocketDisplay/RocketDisplay.gd [hunk 7]
```diff
@@ -1,112 +1,103 @@
...
 	return 2*t*t if t<.5 else -1+(4-2*t)*t
 
-func AnimationFinished(var animation_name):
-	$CameraPivot.SetEnabled(true)
+func animation_finished(var animation_name) -> void:
+	$CameraPivot.set_enabled(true)
 
-func Activate():
+func activate() -> void:
 	$CollisionShape/Display.visible = false
```
Comment (TheDuriel) [inline assets/RocketDisplay/RocketDisplay.gd]: paths. not gonna mark every one. you get the idea.

it makes the lines smaller, and lets us change the path in a single place

File: assets/RocketDisplay/RocketDisplay.gd [hunk 8]
```diff
@@ -1,112 +1,103 @@
...
 	if $AnimationPlayer.is_playing():
 		return
 	current_stage -= 1
-	GoToCurrentStage()
+	go_to_current_stage()
 
-func GoToCurrentStage():
-	if current_stage > 0 and current_stage < max_stages + 1:
+func go_to_current_stage() -> void:
+	if current_stage > 0 and current_stage < MAX_STAGES + 1:
```
Comment (TheDuriel) [inline assets/RocketDisplay/RocketDisplay.gd]: if range(min, max).has(value) perhaps? inbetween range checks are a bit awkward in godot right now. something to put in a util class at a later date.

File: scripts/AthleteController.gd [hunk 9]
```diff
@@ -1,70 +1,71 @@
...
 	
 	if Input.is_action_just_pressed("jump") and $KinematicBody.is_on_wall():
-		jump_force = Vector3(0.0, 150.0, 0.0)
+		_jump_force = Vector3(0.0, 150.0, 0.0)
 		$AnimationTree.set("parameters/Transition/current", "Jump")
 	
-	$KinematicBody.move_and_slide((movement_force * movement_speed * delta) + (jump_force * delta) + (Vector3(0.0, -1.62, 0.0)))
-	movement_force *= 0.95
-	jump_force *= 0.95
+	$KinematicBody.move_and_slide((_movement_force * MOVEMENT_SPEED * delta) + (_jump_force * delta) + (Vector3(0.0, -1.62, 0.0)))
```
Comment (TheDuriel) [inline scripts/AthleteController.gd]: can we break this up into multiple lines? each () block ideally

File: scripts/AvatarPreview.gd [hunk 10]
```diff
@@ -1,51 +1,52 @@
...
+	_skin_mat = base_material.duplicate()
+	_hair_mat = base_material.duplicate()
+	_shoes_mat = base_material.duplicate()
 	
-	$AvatarRig/FemaleRig/Skeleton/AvatarFemale.set_surface_material(1, pants_mat)
-	$AvatarRig/FemaleRig/Skeleton/AvatarFemale.set_surface_material(0, shirt_mat)
-	$AvatarRig/FemaleRig/Skeleton/AvatarFemale.set_surface_material(2, skin_mat)
-	$AvatarRig/FemaleRig/Skeleton/AvatarFemale.set_surface_material(4, hair_mat)
-	$AvatarRig/FemaleRig/Skeleton/AvatarFemale.set_surface_material(3, shoes_mat)
+	$AvatarRig/FemaleRig/Skeleton/AvatarFemale.set_surface_material(1, _pants_mat)
```
Comment (TheDuriel) [inline scripts/AvatarPreview.gd]: this path definitely needs to be put into an onready var

File: scripts/media_mode_camera.gd [hunk 11]
```diff
@@ -56,19 +52,19 @@ func _process(delta):
...
+			look_velocity.x -= event.relative.y * MOUSE_SENSITIVITY * LOOK_ACCELERATION
 	
 	if event.is_action_pressed("scroll_up"):
 		movement_acceleration = min(2.0, movement_acceleration + 0.05)
 		
 	if event.is_action_pressed("scroll_down"):
 		movement_acceleration = max(0.0, movement_acceleration - 0.05)
 
-func _notification(message): 
+func _notification(message) -> void:
```
Comment (TheDuriel) [inline scripts/media_mode_camera.gd]: message is an int

File: assets/RocketDisplay/CameraControl.gd [hunk 12]
```diff
@@ -4,78 +4,76 @@ export (float) var max_zoom_distance : float = 50.0
...
+
+var _current_zoom_distance : float = 20.0
+var _enabled : bool = false
+var _target_rotation_degrees : Vector3 = Vector3()
+var _look_direction : Vector2 = Vector2()
+var _mouse_down : bool = false
 
-func _physics_process(delta) -> void:
-	if not enabled:
+func _physics_process(delta : float) -> void:
```
Comment (TheDuriel) [inline assets/RocketDisplay/CameraControl.gd]:
```var _enabed: bool = false setget set_enabled

func set_enabled(new_value: bool) -> void:
    _enabled = new_value
    set_physics_process(_enabled)
```

File: scripts/AthleteController.gd [hunk 13]
```diff
@@ -49,7 +49,11 @@ func _physics_process(delta) -> void:
 		_jump_force = Vector3(0.0, 150.0, 0.0)
 		$AnimationTree.set("parameters/Transition/current", "Jump")
 	
-	$KinematicBody.move_and_slide((_movement_force * MOVEMENT_SPEED * delta) + (_jump_force * delta) + (Vector3(0.0, -1.62, 0.0)))
+	var movement_velocity = _movement_force * MOVEMENT_SPEED * delta
+	var jump_velocity = _jump_force * delta
+	var gravity_velocity = Vector3(0.0, -1.62, 0.0)
+	$KinematicBody.move_and_slide(movement_velocity + jump_velocity + gravity_velocity)
```
Comment (TheDuriel) [inline scripts/AthleteController.gd]: Cache the body in an onready variable please.

**Comment by TheDuriel** [comment]
You also need to update this branch to masters current state, and resolve the conflicts from that.

https://blog.axosoft.com/learn-git-merge-conflict/ Kraken example.

This is done by merging master into your branch locally, then resolving the conflicts that arise. You can pretty much just take your own version of the file over the one from master.

File: assets/RocketDisplay/CameraControl.gd (hunk 12 above)
Comment (Gyrth) [reply assets/RocketDisplay/CameraControl.gd]: I've eliminated the _enabled variable by using set_physics_process and set_input. Do you agree with this method?

**Comment by Gyrth** [comment]
I've merged master into development and applied my changes. Did I do this correctly?

File: assets/RocketDisplay/CameraControl.gd (hunk 12 above)
Comment (TheDuriel) [reply assets/RocketDisplay/CameraControl.gd]: If its all done locally that works yeah.

**Comment by TheDuriel** [comment]
Looks like everything went well yeah. :D

**Pull request merged**